homework1/task05.py

```diff
@@ -0,0 +1,28 @@
+"""
+Given a list of integers numbers "nums".
+
+You need to find a sub-array with length less equal to "k", with maximal sum.
+
+The written function should return the sum of this sub-array.
+
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+import itertools
+from collections import deque
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    if k <= 0:
+        exit("Think")
+
+    result = []
+    nums = deque(nums)
+
+    while len(nums) > 0:
+        for one in range(1, k + 1):
+            result.append(sum(list(itertools.islice(nums, one))))
```

Comment:
If you use `islice` inside `sum`, you don't have to construct `list`. It should work just fine without intermediate collection construction.